src/dns/aardvark.rs

```diff
@@ -115,6 +115,24 @@ impl Aardvark {
             .env("RUST_LOG", log::max_level().as_str())
             .spawn()?;
 
+        // Starting aardvark server is the last task for netavark so in some env
+        // setup netavark will end up exiting way sooner then aardvark process
+        // is actually ready to serve so after starting aardvark we have to wait
+        // till aardvark process is ready. If it does not shows up atleast retry
+        // 10 times with a delay of 500ms and then return.
+        let mut retry_count = 10;
+        while retry_count > 0 {
+            let aardvark_pid = self.get_aardvark_pid();
+            if aardvark_pid != -1
+                && signal::kill(Pid::from_raw(aardvark_pid), Signal::SIGHUP).is_ok()
```

Comment:
Avoid SIGHUP, send signal 0 if possible - it takes no action on the target process, just returns if the process is alive.